### ISSUE TITLE
ALM stuck when opening an angular CLI project

### ISSUE THREAD

**Issue by martov1** [alm-tools/alm#419]
Hi!,

Alm never finishes loading a project created using angular CLI.
To reproduce simply create an empty angular cli project and try to open it using alm

```
ng new PROJECT_NAME
cd PROJECT_NAME
alm
```

The command line does not indicate any errors and the browser just stays blank
![photo](https://cloud.githubusercontent.com/assets/7854453/22620396/f0bae7be-eae9-11e6-9060-cbd2b218db76.jpg)

**Comment by martov1** [comment]
adding 
```
"include": [
    "src/**/*.ts"
  ]
```

to tslint.json fixes the problem.

**Comment by martov1** [comment]
example project for your convenience https://drive.google.com/file/d/0B-cPCdBZHYeKWmlEaXJQUkpNM28/view?usp=sharing

**Comment by basarat** [comment]
I just ran it in a mac and it worked fine: 

![image](https://cloud.githubusercontent.com/assets/874898/22636578/6ab594d6-ec90-11e6-8be4-2fef7d9e3b0b.png)

Will try on windows later :rose:

**Comment by basarat** [comment]
Your example worked fine on windows as well : 
![image](https://cloud.githubusercontent.com/assets/874898/22640215/c4f51466-eca5-11e6-8185-c63d11bba409.png)

Suspect it was just a momentary thing. Please try again :rose:

**Comment by martov1** [comment]
I still have this problem, i will update chrome and node to see if it gets better.

**Comment by martov1** [comment]
sadly I still have the same problem, is there any way to get a more detailed log on alm?

**Comment by martov1** [comment]
i noticed that the project i sent you does actually open after i updated node, but it takes quite a few minutes to load, I have other larger projects that never quite load and after I close alm with those projects i get the error stacktrace i posted above.

I will upload a large project if you would be so kind to test it out and try to open it.

**Comment by basarat** [comment]
> Does it start working almost immediately for you

yes less than 15s

> is there any way to get a more detailed log on alm

Haven't had the need for it. The dev setup is not that hard : https://basarat.gitbooks.io/alm/content/contributing/ its all TypeScript focused :rose: 



> I will upload a large project if you would be so kind to test it out and try to open it, sorry about the size

for this project it took around a minute to load 
![image](https://cloud.githubusercontent.com/assets/874898/22671207/26c60f44-ed21-11e6-96e2-8bb8ccf20630.png)

Leaving open as I can do some pref work here :rose:

**Comment by basarat** [comment]
Specifically need to figure out why it does TypeScript analysis twice on initial load 

![image](https://cloud.githubusercontent.com/assets/874898/22671230/4d29692e-ed21-11e6-9634-117dcbb4e030.png)

**Comment by basarat** [comment]
> Specifically need to figure out why it does TypeScript analysis twice on initial load

Fixed with `2.27.0` :rose: 

Will try and do more pref work so still leaving open :rose:

**Comment by martov1** [comment]
I have updated, and indeed the typescript analysis starts only once, but on my end, it never ends :/
![photo](https://cloud.githubusercontent.com/assets/7854453/22698223/bd9b549a-ed32-11e6-8706-fe20668f8b32.jpg)

**Comment by basarat** [comment]
Fixed with `2.32.0`. Profiled the UI and it was mostly busy with `setupTree`

![getfilename setup tree](https://cloud.githubusercontent.com/assets/874898/22916019/9adbf12e-f2cf-11e6-99c9-448a2ed0fcdd.png)


So throttled it and its much more responsive. Thanks for your patience :rose: :heart:
